pcsx2/GameDatabase.cpp

```diff
@@ -16,54 +16,91 @@
 #include "PrecompiledHeader.h"
 
 #include "GameDatabase.h"
+#include "Config.h"
+#include "Host.h"
+
+#include "common/FileSystem.h"
+#include "common/Path.h"
+#include "common/StringUtil.h"
+#include "common/Timer.h"
 
 #include "fmt/core.h"
 #include "fmt/ranges.h"
 #include "yaml-cpp/yaml.h"
 #include <fstream>
 #include <algorithm>
+#include <mutex>
+#include <thread>
+
+static constexpr char GAMEDB_YAML_FILE_NAME[] = "GameIndex.yaml";
 
-std::string strToLower(std::string str)
+static std::unordered_map<std::string, GameDatabaseSchema::GameEntry> s_game_db;
+static std::once_flag s_load_once_flag;
+
+static std::string strToLower(std::string str)
 {
 	std::transform(str.begin(), str.end(), str.begin(),
-				   [](unsigned char c) { return std::tolower(c); });
+		[](unsigned char c) { return std::tolower(c); });
 	return str;
 }
 
-bool compareStrNoCase(const std::string str1, const std::string str2)
+static bool compareStrNoCase(const std::string& str1, const std::string& str2)
 {
 	return std::equal(str1.begin(), str1.end(), str2.begin(),
-					  [](char a, char b) {
-						  return tolower(a) == tolower(b);
-					  });
+		[](char a, char b) {
+			return tolower(a) == tolower(b);
+		});
 }
```

Comment:
1. Nitpick: `std::string_view` for arguments may work better.
2. Not nitpick: How does this work when strings are not equal length? `str2.end()` is not used so I'm not sure. Might be worth to safeguard this by comparing lengths first?

Reply:
Unused function from old code.